tests/unit/models/ReplicationConfiguration.spec.ts

```diff
@@ -69,6 +69,143 @@ function getPreferredReadXMLConfig(hasPreferredRead) {
+                        Destination: [{ Bucket: ['arn:aws:s3:::crr-dest'] }],
+                    },
+                ],
+            };
+            const instance = new ReplicationConfiguration({ ReplicationConfiguration: repConfig },
+                null, mockS3ServerConfig);
+            const result = instance.parseConfiguration();
+            expect(result).toBeUndefined();
+            expect(instance.getRole()).toEqual(TEST_ROLE);
+            expect(instance.getDestination()).toEqual('arn:aws:s3:::crr-dest');
+            expect(instance.getRules()).toEqual([
+                {
+                    enabled: true,
+                    id: 'ImagesRule',
+                    prefix: 'main/images/',
+                },
+                {
+                    enabled: false,
+                    id: 'DocsAndProjects',
+                    prefix: 'main/documents/',
+                },
+            ]);
+        });
+
+        it('should return InvalidRequest when config has multiple rules with overlapping prefixes', () => {
+            const repConfig = {
+                Role: [TEST_ROLE],
+                Rule: [
+                    {
+                        ID: ['rule1'],
+                        Prefix: ['prefix/'],
+                        Status: ['Enabled'],
+                        Destination: [{ Bucket: ['arn:aws:s3:::crr-dest'] }],
+                    },
+                    {
+                        ID: ['rule2'],
+                        Prefix: ['prefix/subprefix/'],
+                        Status: ['Enabled'],
+                        Destination: [{ Bucket: ['arn:aws:s3:::crr-dest'] }],
+                    },
+                ],
+            };
+            const instance = new ReplicationConfiguration({ ReplicationConfiguration: repConfig },
+                null, mockS3ServerConfig);
+            const result = instance.parseConfiguration();
+            expect(result).toEqual(errors.InvalidRequest);
+        });
+
+        it('should return InvalidArgument if a prefix is longer than 1024 characters', () => {
+            const repConfig = {
+                Role: [TEST_ROLE],
+                Rule: [{
+                    Prefix: [new Array(1025).fill('X').join('')],
+                    Status: ['Enabled'],
+                    Destination: [{
+                        Bucket: ['arn:aws:s3:::crr-dest'],
+                    }],
+                }],
+            };
+            const instance = new ReplicationConfiguration({ ReplicationConfiguration: repConfig },
+                null, mockS3ServerConfig);
+            const result = instance.parseConfiguration();
+            expect(result).toEqual(errors.InvalidArgument);
+        });
+
+        it('should return MalformedXML if a prefix is an array', () => {
+            const repConfig = {
+                Rule: [
+                    {
+                        ID: ['rule1'],
+                        Prefix: [['']],
```

Comment:
this works in javascript, but I don't see what XML would lead to this : prefix is an array because there could be multiple `<prefix>` tag (possible in XML, though not allowed in this case).

The content of each prefix cannot be an "array", only either a string... or maybe an object (if nested XML tags are used)

test shouldould be added for multiple prefixes though:
```suggestion
                        Prefix: ["foo", "bar"],
```
(i.e. in XML : `<prefix>foo</prefix><prefix>bar</prefix>`)